src/gui/editorwidgets/qgsexternalresourceconfigdlg.cpp

```diff
@@ -107,6 +111,16 @@ void QgsExternalResourceConfigDlg::rootPathPropertyChanged()
   mRootPathButton->setEnabled( !prop.isActive() );
 }
 
+void QgsExternalResourceConfigDlg::documentViewerContentPropertyChanged()
+{
+  QgsProperty prop = mDocumentViewerContentPropertyOverrideButton->toProperty();
+  setDocumentViewerContentExpression( prop );
+
+  mDocumentViewerContentExpression->setVisible( prop.isActive() );
+  //mDocumentViewerContent->setVisible( !prop.isActive() );
+  mDocumentViewerContentComboBox->setEnabled( !prop.isActive() );
```

Comment:
There was (unused) code in the old QgsDataDefinedButton class which automatically handled this use case. I wonder if it's worth resurrecting and copying to QgsPropertyOverrideButton? (see https://github.com/qgis/QGIS/blob/release-2_18/src/gui/qgsdatadefinedbutton.h#L180)